### ISSUE TITLE
Error: One compute environment is expected, but AWS return no compute environment

### ISSUE THREAD

**Issue by BeeShall** [terraform-providers/terraform-provider-aws#12337]
Cannot create an aws batch environment with the following script. Was working perfectly before I made some changes in the name of the environment.

```
resource "aws_batch_compute_environment" "metaflow_batch" {

  compute_environment_name = "${var.namespace}-metaflow-batch"

  compute_resources {
    instance_role = aws_iam_instance_profile.ecs_instance_role.arn

    instance_type = [
      var.instances.type
    ]

    max_vcpus     = var.instances.max_vcpus
    desired_vcpus = var.instances.desired_vcpus
    min_vcpus     = var.instances.min_vcpus

    security_group_ids = [aws_security_group.db_sg.id]

    subnets = var.vpc_config.subnets


    type = "EC2"
  }

  service_role = aws_iam_role.aws_batch_service_role.arn
  type         = "MANAGED"
  depends_on   = [aws_iam_role_policy_attachment.aws_batch_service_role]
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.21
+ provider.aws v2.52.0

### Affected Resource(s)
AWS BATCH

* aws_XXXXX

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Panic Output

Error: One compute environment is expected, but AWS return no compute environment

### Expected Behavior

Batch environment should have been created

### Actual Behavior

Error: One compute environment is expected, but AWS return no compute environment

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by hugadams** [comment]
I've had the same error occur when I removed an AWS Batch compute env. manually and then tried to recreate/refresh it.

**Comment by bflad** [comment]
Fix submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/13935

**Comment by bflad** [comment]
The fix for this has been merged and will be released with version 2.68.0 of the Terraform AWS Provider, likely later today. 👍
